akka-http-core/src/main/scala/akka/http/scaladsl/model/ErrorInfo.scala

```diff
@@ -12,15 +12,45 @@ import StatusCodes.ClientError
  * repeating anything present in the message itself (in order to not open holes for XSS attacks),
  * while the detail can contain additional information from any source (even the request itself).
  */
-final case class ErrorInfo(summary: String = "", detail: String = "") {
+final class ErrorInfo(
```

Comment:
please leave it as a `case class`, no reason to change IMHO; we should mark it as `@InternalApi` perhaps even

Reply:
Thank you for your review.
I put case class back. and add annotation `@InternalAPI` as described below.

```scala
/**
 * INTERNAL API
 * Two-level model of error information.
 * The summary should explain what is wrong with the request or response *without* directly
 * repeating anything present in the message itself (in order to not open holes for XSS attacks),
 * while the detail can contain additional information from any source (even the request itself).
 */
@InternalApi private[akka] final case class ErrorInfo(summary: String = "", detail: String = "", errorHeaderName: String = "") {
  def withSummary(newSummary: String) = copy(summary = newSummary)
  def withSummaryPrepended(prefix: String) = withSummary(if (summary.isEmpty) prefix else prefix + ": " + summary)
  def withErrorHeaderName(headerName: String) = copy(errorHeaderName = headerName.toLowerCase)
  def withFallbackSummary(fallbackSummary: String) = if (summary.isEmpty) withSummary(fallbackSummary) else this
  def formatPretty = if (summary.isEmpty) detail else if (detail.isEmpty) summary else summary + ": " + detail
  def format(withDetail: Boolean): String = if (withDetail) formatPretty else summary
}

/** INTERNAL API */
@InternalApi private[akka] object ErrorInfo {
  /**
   * Allows constructing an `ErrorInfo` from a single string.
   * Used for example when catching exceptions generated by the header value parser, which doesn't provide
   * summary/details information but structures its exception messages accordingly.
   */
  def fromCompoundString(message: String): ErrorInfo = message.split(": ", 2) match {
    case Array(summary, detail) ⇒ apply(summary, detail)
    case _                      ⇒ ErrorInfo("", message)
  }
}
```
I check binary compatible with  `mimaReportBinaryIssues` sbt command.
I still receive a following error log.
```
[error]  * method copy(java.lang.String,java.lang.String)akka.http.scaladsl.model.ErrorInfo in class akka.http.scaladsl.model.ErrorInfo does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.scaladsl.model.ErrorInfo.copy")
[error]  * method this(java.lang.String,java.lang.String)Unit in class akka.http.scaladsl.model.ErrorInfo does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.scaladsl.model.ErrorInfo.this")
[error]  * method apply(java.lang.String,java.lang.String)akka.http.scaladsl.model.ErrorInfo in object akka.http.scaladsl.model.ErrorInfo does not have a correspondent in current version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("akka.http.scaladsl.model.ErrorInfo.apply")
```
Should I add three method(copy, this, apply) to the mima-filters configuration?

Reply:
Yes, the check fail is expected and yes it should be added to the excludes :)

Reply:
akka-http-core/src/main/mima-filters/10.1.0.backwards.excludes

Reply:
Thanks! I understand.
I put case class back. And add some methods to 10.1.0.backwards.excludes.